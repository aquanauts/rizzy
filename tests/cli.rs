use assert_cmd::Command;
use predicates::prelude::*;

#[test]
fn test_timezone_spelling() {
    let assert = Command::cargo_bin("rizzy")
        .unwrap()
        .arg("--zone")
        .arg("America/NewYork")
        .assert();

    assert
        .failure()
        .stderr(predicate::str::contains("America/New_York"));
}

/// Make sure the examples from the README work.
#[test]
fn test_from_readme() {
    let assert = Command::cargo_bin("rizzy")
        .unwrap()
        .arg("--chi")
        .arg("tests/some.log.file")
        .assert();

    let assert = assert.success().stderr(predicate::eq(""));
    let stdout = String::from_utf8(assert.get_output().stdout.clone()).unwrap();
    insta::assert_display_snapshot!(stdout);
}
