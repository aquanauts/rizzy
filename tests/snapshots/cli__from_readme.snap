---
source: tests/cli.rs
assertion_line: 30
expression: stdout

---
# For the integration tests.
2021-09-17T12:28:07-05:00 INFO Running on slurm - limited math CPU usage by a SLURM_CPUS_ON_NODE of 3
2021-09-17T12:30:12-05:00 INFO Initializing Environment. Using Environment.V5_C_PROD

